simpl/rest.py

```diff
@@ -53,3 +57,156 @@ def wrapped(*args):
             return fxn(data, *args)
         return wrapped
     return wrap
+
+
+def paginated(uripath):
+    """Decorator that handles pagination headers, params, and links.
+
+    This accepts, parses, validates, and handles `limit` and `offset` optional
+    query params according to common Rackspace APIs and passes them as kwargs
+    to the decorated function.
+
+        offset: The pagination offset.
+        limit: The pagination limit (or page size). The response is paginated
+            with a default limit of 100 and a maximum limit of 1000.
+
+    Headers returned include:
+        Content-Range: ... per HTTP RFC and 206 response
+        Link: ... one for each page (first, last, next, previous)
+
+    It adds Content-Range headers to the response according to RFC 7233
+    section 4.2 - https://tools.ietf.org/html/rfc7233#section-4.2. The resource
+    type is supplied in the decorator call as uripath.
+
+    It adds link headers based on RFC 5988 (by ex-Racker Mark Nottingham) -
+    https://tools.ietf.org/html/rfc5988#section-5.5.
+
+    Future changes to the RFCs or common API concepts, such as adding links to
+    the response body, may be implemented here to consistently handle
+    pagination across all projects using simpl pagination for their APIs.
+
+    Opinionated assumptions:
+    - body has data items under a `data` or `results` key.
+    - the `len()` of that data represents the number of records being returned
```

Comment:
No, it can be a list or dict (as long as `len()` returns the number of records returned in the call). We're returning `data: [...]` in some projects, which is a slight departure from `results: {...}` in other projects. It might even differ per call.